### ISSUE TITLE
ConfirmationDialog with confirmButtonType danger autofocuses the destructive action

### ISSUE THREAD

**Issue by mattcosta7** [primer/react#2181]
**Describe the bug**

When using a `ConfirmationDialog` with the confirmButtonType danger, the confirmation button is autofocused (rather than the first focusable element or the least destructive action). This makes it likely that users might accidentally perfrom destructive actions, rather than a recoverable one (like canceling)

**Note:** As this is a public repo, please be careful not to include details or screenshots from unreleased GitHub products or features. In most cases, a good bug report should be able to describe the new component without including business logic or feature details, but if you must discuss context relating to an unreleased feature, please open an issue in the private [Design Systems repo](https://github.com/github/design-systems/issues/new/choose) and link to it here.

**To Reproduce**
1. Go to a confirmation dialog with a confirmButtonType=danger
2. Open the dialog
3. note, that the 'confirm' button, which in this case is a dangerous action is autofocus

**Expected behavior**

The least destructive action receives autofocus - or that autofocus behavior is customizable

https://user-images.githubusercontent.com/8616962/180213517-fb3a719c-e314-46bd-acd0-d5bc352a27fb.mov

**Comment by siddharthkp** [comment]
That's great description! 

I see we are explicitly setting autoFocus on the confirmation button, we can set this based on the button type.

```diff
const confirmButton: DialogButtonProps = {
  content: confirmButtonContent,
  buttonType: confirmButtonType,
  onClick: onConfirmButtonClick,
-  autoFocus: true
+  autoFocus: confirmButtonType === 'danger' ? undefined : true
}
```

Focus would not start at the "closest" element, the close button. Should it be on "Cancel" instead?

Happy to accept a pull request. But, before that, can we get the accessibility team's opinion about the default focused element?

**Comment by mattcosta7** [comment]
From a slack thread discussion with `#accessibility`

> For people looking at this thread/providing thoughts, here are mine:
I'm always in two minds about this type of issue.  We can definitely argue that placing focus on the non-destructive button by default increases usability, by decreasing the likelihood of accidents.  However, there are some other things to keep in mind:
What order are the buttons in?  If the non-destructive action is after the destructive one in the focus order, will users know to navigate backwards?
How often do people cancel the destructive action, and how easy is the dialog to trigger in the first place?  If accidentally activating the dialog isn't common, and in 99% of cases people do confirm, are we actually decreasing usability by forcing people to navigate to what is the most common intended destination?
Is this observed in native interfaces?  For example, not sure about macOS, but in Windows the "Yes" button is almost always the default when e.g. deleting a file, restarting your computer, etc.
Is there an undo mechanism?  This might be the most critical; if I delete a file and I can then restore it from somewhere, accidents need not be catastrophic, and therefore these nuances matter a lot less.  But if confirming will literally perform an action that cannot be reversed, then we should probably be doing everything we can to avoid people carrying it out mistakenly.

I think this indicates mostly a need to provide some configuration of the autoFocus behavior. 

a prop like 

```
autoFocus: 'cancel' | 'confirm' = 'confirm'
```

might provide enough configuration to make this easy, without adding much additional complexity?

**Comment by mattcosta7** [comment]
Opened a draft pr - https://github.com/primer/react/pull/2185 for discussion on that proposed api

**Comment by siddharthkp** [comment]
Hi! Read the slack thread and I come up with a different conclusion 😅 

From my understanding, most destructive actions don't have an undo step, so I think we should bake this behavior in if the confirmButtonType is danger and see if we run into a use case where we need to override that.

**Comment by mattcosta7** [comment]
@siddharthkp sounds fine to me.  It should be simple enough later to allow an override prop, with a default that makes sense for the variant

**Comment by mattcosta7** [comment]
Updated pr #2185 based on that feedback
